docs/wiki/EL/el-architecture.md

```diff
@@ -0,0 +1,205 @@
+   - Additional instances of header verification failure can arise when the block numbers are not in sequential order. Typically, the beacon chain is responsible for detecting such discrepancies, although there are instances where it is detected at this stage as well. Failures may also arise when the 1559 base fee is not accurately updated according to the comparison between the last gas used and the gas limit.
+3. Once the header verification is completed, we consider the environment in the header as the environment in which the transactions should be executed and we apply the transactions. We iterate over the transactions in the block and execute each transaction in the evm.
+   - The block headers are passed to the EVM in order to provide the necessary context for processing the transaction. This context includes instructions such as coinbase, gas limit, and timestamp, which are required for proper execution.
+   - Additionally we pass in the transaction and the state
+   - In the event of a failed execution, we simply return the error, indicating an invalid transaction within the block and thereby rendering the block invalid. Within the execution layer, the presence of anything erroneous in a block renders the entire block invalid, as it contaminates the block as a whole.
+   - Once we confirm the validity of the transactions, we proceed to update our state with the result . The state now represent the accumulated state that has all the transaction in the new block applied to it.
+
+From the standpoint of beacon chains, the state transition function mentioned above is encompassed by the invocation of the "new payload" function.
+
+```go
+func newPayload(execPayload engine.ExecutionPayload) bool {
+    if _, err := stf(..); err != nil {
+        return false
+  }
+  return true
+}
+```
+
+The beacon chain  invokes the new payload function and transfers the execution payload as an argument. On the execution layer, we invoke the state transition function using the information from the execution payload. If the state transition function does not produce an error, we return true. Otherwise, we return false to indicate that the block is invalid.
+
+##### Geth
+
+TODO: STF code links and walk through in Geth
+
+##### Sync
+
+TODO
+
+##### Payload building
+
+TODO Link to block-production file
+
+###### Geth
+
+TODO
+
+#### Methods
+
+##### New payload
+
+TODO
+
+###### Geth
+
+TODO
+
+###### Reth
+
+TODO
+
+##### Fork choice updated
+
+TODO
+
+###### Geth
+
+TODO
+
+###### Reth
+
+TODO
+
+### Boot nodes and network bootup
+
+TODO
+
+### Execution layer's blockChain
+
+TODO
+
+### Internal Consensus engines
```

Comment:
Among the client code I was going through they all seem to have a consensus engine component in common and somewhere in some client code, maybe geth, the code comments made a distinction between the External and Internal engines with the external being the Engine API and internal being the engines that offer beacon header verification's, or clique or ethone. I had initially put a table in there following geth's view  on it, I am also a little confused about it but I was going on a leap and guessing that the STF could be broken into many components with the internal consensus engine first. But adding erigon to all the topics is very easy since they have an excellent wiki that I can just provide links to

Reply:
Apologies maybe you were referring to the topic Execution Layer's blockchain.. ah yes I was thinking of erigon and reth's blockchaintree .... but i'll remove that topic for now